src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,42 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private final Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain) throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        HttpSession session = req.getSession();
+        Long driverId = (Long) session.getAttribute("driver_id");
+        if (driverId == null && allowedUrls.contains(req.getServletPath())) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (driverId == null) {
+            resp.sendRedirect(req.getContextPath() + "/login");
+            return;
+        }
```

Comment:
lets refactor it, lets make it just 1 if
```suggestion
        if (driverId == null && !allowedUrls.contains(servletPath)) {
            resp.sendRedirect("/login");
            return;
        }
```